airbyte-integrations/connectors/source-intercom-singer/source_intercom_singer/source.py

```diff
@@ -0,0 +1,81 @@
+"""
+MIT License
+
+Copyright (c) 2020 Airbyte
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in all
+copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
+"""
+
+from airbyte_protocol import AirbyteConnectionStatus, Status
+from base_python import AirbyteLogger, ConfigContainer
+from base_singer import SingerSource
+from tap_intercom.client import IntercomClient, IntercomError
+
+
+class BaseSingerSource(SingerSource):
+    tap_cmd = None
+    tap_name = None
+    api_error = Exception
+
+    def discover_cmd(self, logger: AirbyteLogger, config_path: str) -> str:
+        return f"{self.tap_cmd} --config {config_path} --discover"
+
+    def read_cmd(self, logger: AirbyteLogger, config_path: str, catalog_path: str, state_path: str = None) -> str:
+        args = {"--config": config_path, "--catalog": catalog_path, "--state": state_path}
```

Comment:
Right now with all our singer integrations we are forcing them to be full refresh until we add some singer-specific functionality to allow them to do full refresh. So by default we don't want to pass this integration a state file. Could you add a `force_full_refresh` option which is False by default but if set to true then state is not passed? If that option is set to true as well, then we should implement `discover` like implemented [here](https://github.com/airbytehq/airbyte/pull/1184/files#diff-14093959952a493579c5c2f4f2503427b0f71215fdce401f332a7743e45fb2c5R48)

Reply:
I think you lost me here, do you propose having `force_full_refresh` as a class level option? or as a parameter in methods?

Reply:
@sherifnada I have moved logic to base_singer, please check if that is what you mean